x-pack/plugins/infra/public/alerting/metric_threshold/components/expression.tsx

```diff
@@ -4,25 +4,38 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { debounce } from 'lodash';
+import { debounce, pick } from 'lodash';
+import * as rt from 'io-ts';
+import { HttpSetup } from 'src/core/public';
 import React, { ChangeEvent, useCallback, useMemo, useEffect, useState } from 'react';
 import {
   EuiSpacer,
   EuiText,
   EuiFormRow,
+  EuiButton,
   EuiButtonEmpty,
   EuiCheckbox,
   EuiToolTip,
   EuiIcon,
   EuiFieldSearch,
+  EuiSelect,
+  EuiFlexGroup,
+  EuiFlexItem,
 } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
 import { i18n } from '@kbn/i18n';
 import {
   Comparator,
   Aggregators,
+  METRIC_THRESHOLD_ALERT_TYPE_ID,
   // eslint-disable-next-line @kbn/eslint/no-restricted-paths
 } from '../../../../server/lib/alerting/metric_threshold/types';
+import {
+  INFRA_ALERT_PREVIEW_PATH,
```

Comment:
I think we should move these types to the common directory. That will have the added bonus of allowing us to remove `  // eslint-disable-next-line @kbn/eslint/no-restricted-paths`